ledger/light/executor/state_initer_test.go

```diff
@@ -0,0 +1,249 @@
+	mc := minimock.NewController(t)
+
+	var (
+		jetModifier   *jet.ModifierMock
+		jetReleaser   *JetReleaserMock
+		drops         *drop.ModifierMock
+		nodes         *node.AccessorMock
+		sender        *bus.SenderMock
+		pulseAppender *pulse.AppenderMock
+		pulseAccessor *pulse.AccessorMock
+		jetCalculator *JetCalculatorMock
+		indexes       *object.MemoryIndexModifierMock
+	)
+
+	setup := func() {
+		jetModifier = jet.NewModifierMock(mc)
+		jetReleaser = NewJetReleaserMock(mc)
+		drops = drop.NewModifierMock(mc)
+		nodes = node.NewAccessorMock(mc)
+		sender = bus.NewSenderMock(mc)
+		pulseAppender = pulse.NewAppenderMock(mc)
+		pulseAccessor = pulse.NewAccessorMock(mc)
+		jetCalculator = NewJetCalculatorMock(mc)
+		indexes = object.NewMemoryIndexModifierMock(mc)
+	}
+
+	t.Run("basic error", func(t *testing.T) {
+		setup()
+		nodes = nodes.InRoleMock.Return([]insolar.Node{}, nil)
+
+		s := NewStateIniter(
+			jetModifier,
+			jetReleaser,
+			drops,
+			nodes,
+			sender,
+			pulseAppender,
+			pulseAccessor,
+			jetCalculator,
+			indexes,
+		)
+
+		ref, err := s.heavy(insolar.FirstPulseNumber)
+		assert.Equal(t, *insolar.NewEmptyReference(), ref)
+		assert.Error(t, err, "must return error 'failed to calculate heavy node for pulse'")
+	})
+
+	t.Run("basic ok", func(t *testing.T) {
+		setup()
+		heavy := insolar.NewReference(gen.ID())
+		heavyNodes := []insolar.Node{{*heavy, insolar.StaticRoleHeavyMaterial}}
+		nodes = nodes.InRoleMock.Return(heavyNodes, nil)
+
+		s := NewStateIniter(
+			jetModifier,
+			jetReleaser,
+			drops,
+			nodes,
+			sender,
+			pulseAppender,
+			pulseAccessor,
+			jetCalculator,
+			indexes,
+		)
+
+		ref, err := s.heavy(insolar.FirstPulseNumber)
+		assert.Equal(t, *heavy, ref)
+		assert.NoError(t, err, "must be empty")
+	})
+}
+
+func TestStateIniterDefault_PrepareState(t *testing.T) {
+	ctx := inslogger.TestContext(t)
+	mc := minimock.NewController(t)
+
+	var (
+		jetModifier   *jet.ModifierMock
+		jetReleaser   *JetReleaserMock
+		drops         *drop.ModifierMock
+		nodes         *node.AccessorMock
+		sender        *bus.SenderMock
+		pulseAppender *pulse.AppenderMock
+		pulseAccessor *pulse.AccessorMock
+		jetCalculator *JetCalculatorMock
+		indexes       *object.MemoryIndexModifierMock
+	)
+
+	setup := func() {
+		jetModifier = jet.NewModifierMock(mc)
+		jetReleaser = NewJetReleaserMock(mc)
+		drops = drop.NewModifierMock(mc)
+		nodes = node.NewAccessorMock(mc)
+		sender = bus.NewSenderMock(mc)
+		pulseAppender = pulse.NewAppenderMock(mc)
+		pulseAccessor = pulse.NewAccessorMock(mc)
+		jetCalculator = NewJetCalculatorMock(mc)
+		indexes = object.NewMemoryIndexModifierMock(mc)
+	}
+
+	t.Run("wrong pulse", func(t *testing.T) {
+		setup()
+		s := NewStateIniter(
+			jetModifier,
+			jetReleaser,
+			drops,
+			nodes,
+			sender,
+			pulseAppender,
+			pulseAccessor,
+			jetCalculator,
+			indexes,
+		)
+
+		_, _, err := s.PrepareState(ctx, insolar.FirstPulseNumber/2)
```

Comment:
PulseNumber should never be equal or less than `insolar.FirstPulseNumber`

Reply:
but so this is what the test does, this case is imposible ?